engine.py

```diff
@@ -197,6 +217,84 @@ def log_error(self, msg):
         
     
     ##########################################################################################
+    # panel interfaces
+
+    def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
+        """
+        Shows a panel in a way suitable for this engine. The engine will attempt to
+        integrate it as seamlessly as possible into the host application. If the engine does 
+        not specifically implement panel support, the window will be shown as a modeless
+        dialog instead.
+        
+        :param title: The title of the window
+        :param bundle: The app, engine or framework object that is associated with this window
+        :param widget_class: The class of the UI to be constructed. This must derive from QWidget.
+        
+        Additional parameters specified will be passed through to the widget_class constructor.
+        """
+        # note! not using the import as this confuses nuke's calback system
+        # (several of the key scene callbacks are in the main init file...)
+        import tk_nuke
+        
+        # now look for the tank._callback_from_non_pane_menu property
+        # if this exists, that is an indication that this call comes from 
+        # a non-pane menu, like the Shotgun menu. 
+        #
+        # In this case, we need to explicitly specify where to host the window
+        #
+        try:
+            non_pane_callback = tank._callback_from_non_pane_menu
```

Comment:
I tweaked this to check for the presence of this flag rather than it's value. This way, the menu system can clean up after itself rather than permanently polluting the tank module namespace :)